### PR TITLE
Unify eslint with react-community standard

### PR DESCRIPTION
**Pull request by rborn**
### Does any other open PR do the same thing?

No

### What issue is this PR fixing?

Aligns the eslint/prettier configuration with the `react-community` rules as per https://github.com/react-native-community/.github/issues/9

### How did you test this PR?

`npm run lint`

### CODE REVIEW

File: example/examples/CachedMap.js [hunk 1]
```diff
@@ -18,7 +18,9 @@ class CachedMap extends React.Component {
   constructor(props) {
     super(props);
 
-    const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
+    const ds = new ListView.DataSource({
```
Comment (kelset) [inline example/examples/CachedMap.js]: not in the scope of this PR, but maybe worth migrating away from all ListViews

Comment (kelset) [reply example/examples/CachedMap.js]: (in a separate PR)

File: package.json [hunk 2]
```diff
@@ -39,21 +39,19 @@
...
+    "eslint-plugin-import": "^2.17.2",
+    "eslint-plugin-jsx-a11y": "^6.2.1",
+    "eslint-plugin-prefer-object-spread": "^1.2.1",
+    "eslint-plugin-react": "^7.13.0",
     "gitbook-cli": "^2.3.0",
     "lodash": "^4.17.2",
     "metro-react-native-babel-preset": "^0.53.1",
-    "prop-types": "^15.7.2",
-    "react": "16.8.3",
+    "react": "^16.8.6",
```
Comment (kelset) [inline package.json]: I think you want to stay on `16.8.3` fixed

File: package.json [hunk 3]
```diff
@@ -39,21 +39,19 @@
...
     "babel-plugin-module-resolver": "^3.2.0",
-    "babel-preset-airbnb": "^1.1.1",
-    "babel-preset-react-native": "1.9.0",
-    "eslint": "^3.3.1",
-    "eslint-config-airbnb": "^10.0.1",
-    "eslint-plugin-import": "^1.14.0",
-    "eslint-plugin-jsx-a11y": "^2.1.0",
-    "eslint-plugin-prefer-object-spread": "^1.1.0",
-    "eslint-plugin-react": "^6.1.2",
+    "babel-preset-react-native": "4.0.1",
```
Comment (kelset) [inline package.json]: I think you don't need this directly anymore, nor most of the other plugins as they are not used in the eslint config

File: example/examples/CachedMap.js (hunk 1 above)
Comment (rborn) [reply example/examples/CachedMap.js]: I'll try to make a PR for that, thnx

**Comment by termux-io** [comment]
language: ruby
dist: trusty
cache:
  bundler: true
  directories:
    - node_modules
sudo: required

addons:
  chrome: stable

before_install:
  - google-chrome-stable --headless --disable-gpu --remote-debugging-port=9222 http://localhost &

services:
  - postgresql

before_script:
  - cp config/database.yml.travis config/database.yml
  - psql -c 'create database travis_ci_test;' -U postgres

script:
  - bundle exec bin/setup_travis
  - bundle exec rake travis:ci
env:
  global:
    - CI_NODE_TOTAL=4
  matrix:
    - CI_NODE_INDEX=0
    - CI_NODE_INDEX=1
    - CI_NODE_INDEX=2
    - CI_NODE_INDEX=3
    - JASMINE=true
branches:
  only: # Only build master and e2e-services. All pull requests also get built
    - master
    - e2e-services
deploy:
  provider: script
  script: bin/ecc-sync
  on:
    branch: master
    repo: nasa/earthdata-search

**Pull request merged**